atd-vzd/triggers/update_noncr3_location.sql

```diff
@@ -0,0 +1,40 @@
+-- Function for atd_apd_blueform generated column called location_id
+CREATE OR REPLACE FUNCTION update_noncr3_location(blueform_case_id integer)
+  RETURNS varchar 
+  LANGUAGE PLPGSQL
+  IMMUTABLE
+  AS
+$$
+BEGIN
+    -- Check if crash is main-lane and of concern to TxDOT
+	IF EXISTS (SELECT atc.*
+                FROM atd_apd_blueform AS atc
+                INNER JOIN non_cr3_mainlanes AS ncr3m ON (
+                atc.position && ncr3m.geometry
+                AND ST_Contains(
+                        ST_Transform(
+                                ST_Buffer(
+                                    ST_Transform(ncr3m.geometry, 2277),
+                                    1,
+                                    'endcap=flat join=round'
+                                ),
+                                4326
+                        ), /* transform into 2277 to buffer by a foot, not a degree */
+                        atc.position
```

Comment:
i think if you pass in the position of the crash as a parameter to this function, you could just make this a select without having to do a join. like
`IF EXISTS (SELECT * FROM non_cr3_mainlanes WHERE (position && non_cr3_mainlanes.geometry) AND ....`

Reply:
oh nice - i definitely just copied/pasted without looking at these queries too much 😝 I'll do a pass over these. i like the suggestions - thanks! 🙏